public/app/features/auth-config/AuthConfigPage.tsx

```diff
@@ -43,14 +42,18 @@ export const AuthConfigPageUnconnected = ({ providerStatuses, isLoading, loadSet
   }, [loadSettings]);
 
   const authProviders = getRegisteredAuthProviders();
-  const enabledProviders = authProviders.filter((p) => providerStatuses[p.id]?.enabled);
-  const configuresProviders = authProviders.filter(
-    (p) => providerStatuses[p.id]?.configured && !providerStatuses[p.id]?.enabled
-  );
-  const availableProviders = authProviders.filter(
-    (p) => !providerStatuses[p.id]?.enabled && !providerStatuses[p.id]?.configured
-  );
-  const firstAvailableProvider = availableProviders?.length ? availableProviders[0] : null;
+  const checkProviderStatus = (provider: AuthProviderInfo): boolean => {
+    // if it is configured in the UI, that takes precidence it is enabled in the UI
+    // might want to refactor to just have a enabled field from the backend that is specified
+    if (providerStatuses[provider.id]?.configuredInUI) {
+      return providerStatuses[provider.id]?.configuredInUI || providerStatuses[provider.id]?.enabledInUI;
+    }
```

Comment:
What exactly are you trying to check?
This will always return `true` is `configuredInUI` is `true`.

Reply:
this checks if `providerStatuses[provider.id]?.configuredInUI is truthy` (i.e., it exists and is not false, 0, "", null, undefined, or NaN).
Therefore we can compare it and also evaluate if if it enabled

Reply:
exists as a configuration at all. trying to simplifying the logic here as much as possible that's why. it was a mess before and it is a mess now

Reply:
I see you've made the change :slightly_smiling_face: :+1: